### ISSUE TITLE
Error when using nix-shell, looking for help

### ISSUE THREAD

**Issue by feiyeung** [vlaci/openconnect-sso#65]
Hi, I'm trying to build the project, and i just had a fresh install of nix. I haven't used nix before.

I got this error when launching `nix-shell` from the repo directory. I'm not sure if that's an issue with nix itself or something related to this project. Any help or pointer is appreciated. Thanks!

```
$ nix-shell  --show-trace
error: while evaluating the attribute 'buildInputs' of the derivation 'nix-shell' at /nix/store/pjyp4bcyh0mh7q200zlpq2isd3h6r4d3-nixpkgs-21.11pre317548.e0ce3c683ae/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:205:7:
while evaluating the attribute 'out.outPath' at /nix/store/pjyp4bcyh0mh7q200zlpq2isd3h6r4d3-nixpkgs-21.11pre317548.e0ce3c683ae/nixpkgs/lib/customisation.nix:156:13:
while evaluating the attribute 'buildInputs' of the derivation 'python3.9-poetry-1.1.4' at /nix/store/pjyp4bcyh0mh7q200zlpq2isd3h6r4d3-nixpkgs-21.11pre317548.e0ce3c683ae/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:205:7:
while evaluating the attribute 'out.outPath' at /nix/store/pjyp4bcyh0mh7q200zlpq2isd3h6r4d3-nixpkgs-21.11pre317548.e0ce3c683ae/nixpkgs/lib/customisation.nix:156:13:
while evaluating the attribute 'buildInputs' of the derivation 'python3.9-poetry-core-1.0.0' at /nix/store/pjyp4bcyh0mh7q200zlpq2isd3h6r4d3-nixpkgs-21.11pre317548.e0ce3c683ae/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:205:7:
while evaluating 'getOutput' at /nix/store/pjyp4bcyh0mh7q200zlpq2isd3h6r4d3-nixpkgs-21.11pre317548.e0ce3c683ae/nixpkgs/lib/attrsets.nix:489:23, called from undefined position:
while evaluating anonymous function at /nix/store/pjyp4bcyh0mh7q200zlpq2isd3h6r4d3-nixpkgs-21.11pre317548.e0ce3c683ae/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:156:13, called from undefined position:
while evaluating the attribute 'setuptools-scm' at /nix/store/p4giqc2yspwbf7aq95lmnaxdxpizqi12-source/default.nix:211:19:
while evaluating the attribute 'setuptools_scm' at undefined position:
while evaluating anonymous function at /nix/store/pjyp4bcyh0mh7q200zlpq2isd3h6r4d3-nixpkgs-21.11pre317548.e0ce3c683ae/nixpkgs/pkgs/top-level/python-aliases.nix:27:22, called from undefined position:
while evaluating 'removeDistribute' at /nix/store/pjyp4bcyh0mh7q200zlpq2isd3h6r4d3-nixpkgs-21.11pre317548.e0ce3c683ae/nixpkgs/pkgs/top-level/python-aliases.nix:15:22, called from /nix/store/pjyp4bcyh0mh7q200zlpq2isd3h6r4d3-nixpkgs-21.11pre317548.e0ce3c683ae/nixpkgs/pkgs/top-level/python-aliases.nix:27:29:
while evaluating 'isDerivation' at /nix/store/pjyp4bcyh0mh7q200zlpq2isd3h6r4d3-nixpkgs-21.11pre317548.e0ce3c683ae/nixpkgs/lib/attrsets.nix:330:18, called from /nix/store/pjyp4bcyh0mh7q200zlpq2isd3h6r4d3-nixpkgs-21.11pre317548.e0ce3c683ae/nixpkgs/pkgs/top-level/python-aliases.nix:16:8:
while evaluating 'removeRecurseForDerivations' at /nix/store/pjyp4bcyh0mh7q200zlpq2isd3h6r4d3-nixpkgs-21.11pre317548.e0ce3c683ae/nixpkgs/pkgs/top-level/python-aliases.nix:8:33, called from /nix/store/pjyp4bcyh0mh7q200zlpq2isd3h6r4d3-nixpkgs-21.11pre317548.e0ce3c683ae/nixpkgs/pkgs/top-level/python-aliases.nix:28:31:
while evaluating 'checkInPkgs' at /nix/store/pjyp4bcyh0mh7q200zlpq2isd3h6r4d3-nixpkgs-21.11pre317548.e0ce3c683ae/nixpkgs/pkgs/top-level/python-aliases.nix:22:20, called from /nix/store/pjyp4bcyh0mh7q200zlpq2isd3h6r4d3-nixpkgs-21.11pre317548.e0ce3c683ae/nixpkgs/pkgs/top-level/python-aliases.nix:29:32:
infinite recursion encountered, at /nix/store/pjyp4bcyh0mh7q200zlpq2isd3h6r4d3-nixpkgs-21.11pre317548.e0ce3c683ae/nixpkgs/pkgs/top-level/python-aliases.nix:76:20
```

**Comment by jbedo** [comment]
This looks like #68. It seems to work with the 21.05 channel so I suggest adding `-I nixpkgs=channel:nixos-21.05` to your `nix-shell` command and see how you go.

**Comment by vlaci** [comment]
Should be fixed on 0.8.0. Unfortunately I cannot test it. Feel free to reopen if the issue persist.
